main.c

```diff
@@ -30,30 +43,147 @@ static void lv_linux_disp_init(void)
 #elif LV_USE_SDL
 static void lv_linux_disp_init(void)
 {
-    const int width = atoi(getenv("LV_SDL_VIDEO_WIDTH") ? : "800");
-    const int height = atoi(getenv("LV_SDL_VIDEO_HEIGHT") ? : "480");
 
-    lv_sdl_window_create(width, height);
+    lv_sdl_window_create(window_width, window_height);
+
+}
+#elif LV_USE_WAYLAND
+static void lv_linux_disp_init(void)
+{
+    lv_display_t *disp;
+    lv_group_t *g;
+
+    lv_wayland_init();
+    disp = lv_wayland_window_create(window_width, window_height, "LVGL Simulator", NULL);
+
+    if (fullscreen) {
+            lv_wayland_window_set_fullscreen(disp, fullscreen);
+    } else if (maximize) {
+            lv_wayland_window_set_maximized(disp, maximize);
+    }
+
+    g = lv_group_create();
+    lv_group_set_default(g);
+    lv_indev_set_group(lv_wayland_get_keyboard(disp), g);
+    lv_indev_set_group(lv_wayland_get_pointeraxis(disp), g);
+
 }
 #else
 #error Unsupported configuration
 #endif
 
-int main(void)
+#if !LV_USE_WAYLAND
+
+void default_run_loop(void)
+{
+    /*Handle LVGL tasks*/
+    while(1) {
+        lv_timer_handler();
+        usleep(5000);
+    }
+}
+
+
+#else
+
+void wayland_run_loop(void)
+{
+    struct pollfd pfd;
+    int sleep_wait = 0;
+    uint32_t time_till_next;
+    uint8_t tickless = 0;
+
+    pfd.fd = lv_wayland_get_fd();
+    pfd.events = POLLIN;
+
+    /* Handle LVGL tasks */
+    while (1) {
+
+        if (tickless == 0 && sleep_wait > 0) {
+            lv_tick_inc(sleep_wait);
+        }
+
+        /* Handle any Wayland/LVGL timers/events */
+        time_till_next = lv_wayland_timer_handler();
+
+        /* Run until the last window closes */
+        if (!lv_wayland_window_is_open(NULL)) {
+            break;
+        }
+
+        /* Wait for something for an event to happen */
+        if (time_till_next == LV_NO_TIMER_READY) {
+            sleep_wait = -1;
+        } else if (time_till_next > INT_MAX) {
+            sleep_wait = INT_MAX;
+        } else {
+            sleep_wait = time_till_next;
+        }
+
+        while ((poll(&pfd, 1, sleep_wait) < 0) && (errno == EINTR));
+    }
+}
+
+#endif
+
+int main(int argc, char **argv)
 {
+    int opt = 0;
+    bool err = false;
+
+    /* Default values */
+    fullscreen = maximize = false;
```

Comment:
Move the config part it to function for better readibility.

Reply:
Do you mean move the switch/case used to get the options to a dedicated function ?  Yes sure can

Reply:
Yes, jsut to keep `main()` easier to read.